components/scream/src/physics/shoc/shoc_functions_f90.cpp

```diff
@@ -2269,22 +2269,29 @@ void update_prognostics_implicit_f(Int shcol, Int nlev, Int nlevi, Int num_trace
     v_wind_d(temp_2d_d[12]);
 
   view_3d
-    tracer_d(temp_3d_d[0]);
-
-  // update_prognostics_implicit() expects 3 extra tracer slots
-  Kokkos::resize(tracer_d,
-                 shcol,nlev,ekat::npack<Spack>(num_tracer+3));
+    tracer_f90_d(temp_3d_d[0]);
 
   // Local variables
   const Int nlev_packs = ekat::npack<Spack>(nlev);
   const Int nlevi_packs = ekat::npack<Spack>(nlevi);
   const auto policy = ekat::ExeSpaceUtils<ExeSpace>::get_default_team_policy(shcol, nlev_packs);
 
-  view_3d
-    X1_d("X1",shcol,nlev,ekat::npack<Spack>(2));
+  // CXX version of shoc qtracers is the transpose of the fortran version.
+  view_3d tracer_cxx_d("",shcol,num_tracer,nlev_packs);
+  Kokkos::parallel_for(policy, KOKKOS_LAMBDA(const MemberType& team) {
+    const int i = team.league_rank();
+    Kokkos::parallel_for(Kokkos::TeamThreadRange(team, nlev), [&] (const Int& k) {
+      Kokkos::parallel_for(Kokkos::ThreadVectorRange(team, num_tracer), [&] (const Int& q) {
+        tracer_cxx_d(i,q,k/Spack::n)[k%Spack::n] = tracer_f90_d(i,k,q/Spack::n)[q%Spack::n];
+      });
+    });
+  });
```

Comment:
Fortran tracers are now the transpose of C++ tracers. I just transpose the data for the bfb tests before and after calling the function here and in `shoc_main_f()`.